airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/helpers.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import re
+from typing import List
+
+from pygsheets import Spreadsheet, Worksheet
+from pygsheets.exceptions import WorksheetNotFound
+
+
+def get_spreadsheet_id(id_or_url: str) -> str:
+    if re.match(r"(http://)|(https://)", id_or_url):
+        m = re.search(r"(/)([-\w]{40,})([/]?)", id_or_url)
+        if m.group(2):
+            return m.group(2)
+    else:
+        return id_or_url
```

Comment:
surely if we don't match it's _possible_ that we've got an invalid id or url, we could at least log a warning here to give an indication that this _might_ be an issue.

Reply:
Done.